### PR TITLE
W-10673821: Add move-to-folder operation.

### PR DESCRIPTION
**Pull request by giss-ignacio**

### CODE REVIEW

File: src/main/java/org/mule/extension/email/internal/mailbox/imap/IMAPOperations.java [hunk 1]
```diff
@@ -174,4 +185,52 @@ public void delete(@Connection MailboxConnection connection,
...
+      if (!destinationFolder.exists()) {
+        if (createTargetFolder) {
+          destinationFolder.create(Folder.HOLDS_FOLDERS | Folder.HOLDS_MESSAGES);
+        } else {
+          throw new FolderNotFoundException(destinationFolder);
+        }
+      }
+
+      IMAPFolder sourceMailboxFolder = (IMAPFolder) connection.getFolder(sourceFolder, READ_WRITE);
+      javax.mail.Message message = sourceMailboxFolder.getMessageByUID(emailId);
```
Comment (konangc87) [inline src/main/java/org/mule/extension/email/internal/mailbox/imap/IMAPOperations.java:219]: Could the sourceMailboxFolder be null? if so, a null check should be in place and throw an FolderNotFoundException on sourceFolder.

**Review by konangc87 — CHANGES_REQUESTED** [review]

File: src/main/java/org/mule/extension/email/internal/mailbox/imap/IMAPOperations.java [hunk 2]
```diff
@@ -11,6 +11,12 @@
 import static javax.mail.Flags.Flag.DELETED;
 import static javax.mail.Flags.Flag.SEEN;
 import static javax.mail.Folder.READ_WRITE;
+
+import com.sun.mail.imap.IMAPFolder;
+import org.mule.extension.email.api.exception.EmailAccessingFolderException;
+import org.mule.extension.email.api.exception.EmailException;
```
Comment (nicomz) [inline src/main/java/org/mule/extension/email/internal/mailbox/imap/IMAPOperations.java:17]: Unused Import

File: src/main/java/org/mule/extension/email/internal/mailbox/imap/IMAPOperations.java [hunk 3]
```diff
@@ -47,6 +53,11 @@
 import org.mule.runtime.extension.api.runtime.streaming.PagingProvider;
 import org.mule.runtime.extension.api.runtime.streaming.StreamingHelper;
 
+import javax.mail.Folder;
+import javax.mail.FolderNotFoundException;
+import javax.mail.Message;
+import javax.mail.UIDFolder;
```
Comment (nicomz) [inline src/main/java/org/mule/extension/email/internal/mailbox/imap/IMAPOperations.java:59]: Unused Import

**Review by nicomz — CHANGES_REQUESTED** [review]

File: src/test/java/org/mule/email/mtf/TestIMAPServer.java [hunk 4]
```diff
@@ -48,6 +53,7 @@ public static void start(Integer port) {
   }
 
   public static void startSecure(Integer port) {
+    Security.setProperty("ssl.SocketFactory.provider", DummySSLSocketFactory.class.getName());
```
Comment (nicomz) [inline src/test/java/org/mule/email/mtf/TestIMAPServer.java:56]: Revert to the original value on stop.

**Review by nicomz — CHANGES_REQUESTED** [review]

**Review by konangc87 — APPROVED** [review]

**Review by nicomz — APPROVED** [review]

**Pull request merged**